src/kernel/arch/aarch64/arch.zig

```diff
@@ -120,5 +109,133 @@ pub fn haltNoInterrupts() noreturn {
 
 // TODO: implement
 pub fn spinWait() noreturn {
-    while (true) {}
+    spinLed(500);
+}
+
+pub inline fn turnOnLed() void {
+    asm volatile (
+        \\ mov x0, #0x3f200000
+        \\ ldr w1, [x0, #0x08]
+        \\ mov x2, #0x38000000
+        \\ bic x1, x1, x2
+        \\ mov x2, #0x08000000
+        \\ orr x1, x1, x2
+        \\ str w1, [x0, #0x08]
+        \\ mov x1, #0x20000000
+        \\ str w1, [x0, #0x1c]
+        :
+        :
+        : "x0", "x1", "x2"
+    );
+}
+
+pub inline fn turnOffLed() void {
+    asm volatile (
+        \\ mov x0, #0x3f200000
+        \\ ldr w1, [x0, #0x08]
+        \\ mov x2, #0x38000000
+        \\ bic x1, x1, x2
+        \\ mov x2, #0x08000000
+        \\ orr x1, x1, x2
+        \\ str w1, [x0, #0x08]
+        \\ mov x1, #0x20000000
+        \\ str w1, [x0, #0x28]
+        :
+        :
+        : "x0", "x1", "x2"
+    );
+}
```

Comment:
Could these use more zig code instead? We like to keep raw assembly to the minimum.

Reply:
Yes. I wanted to get your eyes on this before proceeding further.The very point of updating tcr and ttbr0 is delicate and will probably still be asm copied from the arm manual.

Reply:
These are the instructions for updating tcr and ttbr0. This is the 1.0 manual. There is a 1.1 manual but it does not have an online link - the 1.1 version can only be used locally. This particukar section of the manual is the same.

https://static.docs.arm.com/100940/0100/armv8_a_address%20translation_100940_0100_en.pdf#%5B%7B%22num%22%3A53%2C%22gen%22%3A0%7D%2C%7B%22name%22%3A%22XYZ%22%7D%2C54%2C456%2C0%5D

Reply:
> Yes. I wanted to get your eyes on this before proceeding further. Updating tcr and ttbr0 is delicate and will probably still be asm copied from the arm manual.

Sure, if something has to use inline assembly that's fine. Register moves, arithmetic and the like should preferably use Zig.